R/score_model_out.R

```diff
@@ -0,0 +1,247 @@
+#' Score model output predictions with a single `output_type` against observed data
+#'
+#' @param model_out_tbl Model output tibble with predictions
+#' @param target_observations Observed 'ground truth' data to be compared to
+#' predictions
+#' @param metrics Optional list of scoring metrics to compute. See details for more.
+#' @param summarize boolean indicator of whether summaries of forecast scores
+#' should be computed. Defaults to `TRUE`.
+#' @param by character vector naming columns to summarize by. For example,
+#' specifying `by = "model_id"` (the default) will compute average scores for
+#' each model.
+#' @param output_type_id_order For ordinal variables in pmf format, this is a
+#' vector of levels for pmf forecasts, in increasing order of the levels. For
+#' all other output types, this is ignored.
+#'
+#' @details If `metrics` is `NULL` (the default), this function chooses
+#' appropriate metrics based on the `output_type` contained in the `model_out_tbl`:
+#' - For `output_type == "quantile"`, we use the default metrics provided by
+#' `scoringutils::metrics_quantile()`: "wis", "overprediction", "underprediction",
```

Comment:
This is a cool suggestion r.e. generating the documentation dynamically.  Tragically, it didn't work when I tried it, for reasons I couldn't figure out (in part because I still have no idea how to dig into errors that seem to be reported by cli/rlang) :sob:.  Here is what I got:

```
> devtools::document()
ℹ Updating hubEvals documentation
ℹ Loading hubEvals
✖ score_model_out.R:16: @details failed to evaluate inline markdown code.
Caused by error in `if (...) NULL`:
! missing value where TRUE/FALSE needed
```
No actual error was thrown and I don't know where this `if (...) NULL` is coming from.  `traceback()` and various `rlang` functions I tried were unhelpful.  :shrug:

Reply:
The list needs to be indented for this to work: https://github.com/r-lib/roxygen2/issues/1651

I found the error from the source code and ran `trace(roxygen2:::markdown, tracer = quote(cat(sprintf("text = %s\n", text))))` to reveal the context of the failure.

Reply:
Interesting... great find!

Reply:
Agreed, thanks for the investigation @zkamvar!  In https://github.com/hubverse-org/hubEvals/pull/46/commits/285b4d995408e7e73eae85be404c3c72e1ac08a5 I used a bit of a sad workaround with `\itemize` to get this going.